### PR TITLE
merge leftovers from `clean-up`

### PR DESCRIPTION
**Pull request by marinlauber**
All my fix and tests were actually never in the origin pull request of `clean-up`... I think I pushed to `origin` instead of `upstream` 🤦.

There it is then. This implements
- adding some test for `integrals.jl`
- fixing type stability for `_pforce` and `_vforce`
- add check to avoid integrating forces on bodies that are not 2D curves.

### CODE REVIEW

**Comment by marinlauber** [comment]
This is ready, all the test run as well.

File: example/Project.toml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 [deps]
 CUDA = "052768ef-5323-5732-b1bb-66c8b64840ba"
 GLMakie = "e9467ef8-e4e7-5192-8a1a-b1aee30e663a"
+Makie = "ee78f7c6-11fb-53f2-987a-cfe4a2b5a57a"
```
Comment (marinlauber) [inline example/Project.toml:4]: Makie is not needed here now actually.

Comment (weymouth) [reply example/Project.toml:4]: We should probably have a 3D example what uses Makie and Meshing. I'll think about that a little.

Comment (marinlauber) [reply example/Project.toml:4]: All of them can be run with Makie, we have an sdf that we can mesh.

**Pull request merged**